src/solvers/dgsem_tree/dg_2d_subcell_limiters.jl

```diff
@@ -26,30 +26,33 @@ function create_cache(mesh::TreeMesh{2}, equations,
                                        nnodes(dg) + 1) for _ in 1:Threads.nthreads()]
     flux_temp_threaded = A3d[A3d(undef, nvariables(equations), nnodes(dg), nnodes(dg))
                              for _ in 1:Threads.nthreads()]
-    flux_nonconservative_temp_threaded = A4d[A4d(undef, nvariables(equations),
-                                                 nnoncons(equations),
-                                                 nnodes(dg), nnodes(dg))
-                                             for _ in 1:Threads.nthreads()]
     fhat_temp_threaded = A3d[A3d(undef, nvariables(equations), nnodes(dg),
                                  nnodes(dg))
                              for _ in 1:Threads.nthreads()]
-    fhat_nonconservative_temp_threaded = A4d[A4d(undef, nvariables(equations),
-                                                 nnoncons(equations),
-                                                 nnodes(dg), nnodes(dg))
-                                             for _ in 1:Threads.nthreads()]
-
-    phi_threaded = A4d[A4d(undef, nvariables(equations),
-                           nnoncons(equations),
-                           nnodes(dg), nnodes(dg))
-                       for _ in 1:Threads.nthreads()]
-
     antidiffusive_fluxes = Trixi.ContainerAntidiffusiveFlux2D{uEltype}(0,
                                                                        nvariables(equations),
                                                                        nnodes(dg))
+
+    if typeof(have_nonconservative_terms(equations)) == True
```

Comment:
```suggestion
    if have_nonconservative_terms(equations)
```
isn't this sufficient?

Reply:
No, `have_nonconservative_terms(equations)` does not return a variable of type `Bool` (e.g., `true` or `false`), but a variable of type `True`. It is commonly used for multiple dispatch.

Reply:
Ah, right, it is of Static.jl. Thus you can do the following:
```suggestion
    if have_nonconservative_terms(equations) == true
```
This should work since they define something like `==(a::Bool, b::False)` for you, and it is a little easier to understand